File: Python/dsaOne/Blind 75/Javascript/Arrays_and_Hashing/containsDuplicate.js
//https://leetcode.com/problems/contains-duplicate/description/

const containsDuplicate = (nums) => {
    let hm = {}
    for (let i = 0; i < nums.length; i++) {
        if(nums[i] in hm){
            return true
        }
        else {
            hm[nums[i]] = 1
        }
    }
    return false
}

console.log(containsDuplicate([1,2,3,1]))
console.log(containsDuplicate([1,2,3,4]))

File: Python/dsaOne/Blind 75/arrays_and_hashing./containsDuplicate.py
#https://leetcode.com/problems/contains-duplicate/

def containsDuplicate(nums):
    hm = {}
    for i in nums:
        if i in hm:
            return True
        else:
            hm[i] = 1
    return False
    #return len(nums) != len(set(nums))

print(containsDuplicate([1,2,3,1]))
print(containsDuplicate([1,2,3,4]))